sauna/src/controllers/users.js

```diff
@@ -0,0 +1,78 @@
+import express from 'express';
+import crypto from 'crypto';
+import mongoose from 'mongoose';
+
+import User from '../models/user.js';
+
+const router = express.Router();
+
+const countTotalTime = (u) => u.editedProjects.reduce((a, c) => (a + c.totalTime), 0);
+
+/** Our global leaderboard. */
+router.get('/leaderboard', async (req, res) => {
+    const users = await User.find({})
+        .populate('editedProjects');
+
+    const sortedUsers = users.sort((a, b) => countTotalTime(b) - countTotalTime(a)).splice(0, 10);
```

Comment:
not familiar with Mongo, but could this be done on database side with some sort of index? this will get expensive quickly. If not it should be cached for some period of time